src/main/java/com/exasol/projectkeeper/validators/dependencies/renderer/DependencyPageRenderer.java

```diff
@@ -0,0 +1,79 @@
+        reportBuilder
+                .append(buildDependencySectionForScope(dependencies, Dependency.Scope.TEST, markdownReferenceBuilder));
+        reportBuilder.append(
+                buildDependencySectionForScope(dependencies, Dependency.Scope.RUNTIME, markdownReferenceBuilder));
+        reportBuilder.append(
+                buildDependencySectionForScope(dependencies, Dependency.Scope.PLUGIN, markdownReferenceBuilder));
+        reportBuilder.append("\n");
+        reportBuilder.append(markdownReferenceBuilder.getReferences());
+        return reportBuilder.toString();
+    }
+
+    private String buildDependencySectionForScope(final List<Dependency> dependencies, final Dependency.Scope scope,
+            final MarkdownReferenceBuilder markdownReferenceBuilder) {
+        final List<Dependency> dependenciesOfThisScope = dependencies.stream()
+                .filter(dependency -> dependency.getScope().equals(scope)).collect(Collectors.toList());
+        if (dependenciesOfThisScope.isEmpty()) {
+            return "";
+        } else {
+            final String heading = "## " + capitalizeFirstLetter(scope.name()) + " Dependencies";
+            return "\n" + heading + "\n\n" + buildTable(dependenciesOfThisScope, markdownReferenceBuilder) + "\n";
+        }
+    }
+
+    private String buildTable(final List<Dependency> dependencies,
+            final MarkdownReferenceBuilder markdownReferenceBuilder) {
+        final Table.Builder tableBuilder = new Table.Builder().withAlignments(Table.ALIGN_LEFT, Table.ALIGN_LEFT);
+        tableBuilder.addRow("Dependency", "License");
+        dependencies.forEach(dependency -> {
+            final String name = renderDependencyName(dependency, markdownReferenceBuilder);
+            final String licenseText = dependency.getLicenses().stream()
+                    .map(license -> renderLicense(license, markdownReferenceBuilder)).collect(Collectors.joining("; "));
+            tableBuilder.addRow(name, licenseText);
+        });
+        return tableBuilder.build().toString();
+    }
+
+    private String renderDependencyName(final Dependency dependency,
+            final MarkdownReferenceBuilder markdownReferenceBuilder) {
+        return renderLink(dependency.getName(), dependency.getUrl(), markdownReferenceBuilder);
+    }
+
+    private String renderLicense(final License license, final MarkdownReferenceBuilder markdownReferenceBuilder) {
+        return renderLink(license.getName(), license.getUrl(), markdownReferenceBuilder);
+    }
+
+    private String renderLink(final String name, final String url,
+            final MarkdownReferenceBuilder markdownReferenceBuilder) {
+        if (url == null || url.isBlank()) {
+            return name;
+        } else {
+
+            return "[" + name + "][" + markdownReferenceBuilder.getReferenceForUrl(name, url) + "]";
+        }
+    }
+
+    private String capitalizeFirstLetter(final String string) {
```

Comment:
I would place this method in the Scope class

Reply:
I think it's better placed here because it's specific for the rendering.

Reply:
Well, it's not really critical, just would be more elegant

Reply:
I agree, it would be a nicer code but breaking separation of concerns.